### ISSUE TITLE
Feature Request: autoload webcomponents.js and dart_support.js for non-chrome browsers

### ISSUE THREAD

**Issue by monkeysecurity** [sethladd/dart_to_js_script_rewriter#6]
```test.dart.js``` could detect whether the browser supports the ShadowDom and if not, include the following scripts:

    <script async src="packages/web_components/webcomponents.min.js"></script>
    <script async src="packages/web_components/dart_support.js"></script>

The scripts would then be skipped in browsers supporting ShadowDom, which should speed up load time.

Let me know if this sounds like a reasonable idea.

**Comment by sethladd** [comment]
where does test.dart.js come from?

**Comment by monkeysecurity** [comment]
Oh, sorry for not being clear.

A traditional dart application is deployed with something like this:

    <script async type="application/dart" src="test.dart"></script>
    <script async src="packages/browser/dart.js"></script>
    
    
    <script async src="packages/web_components/webcomponents.min.js"></script>
    <script async src="packages/web_components/dart_support.js"></script>

This transformer currently changes this to:

    <script async src="test.dart.js"></script>
    
    
    <script async src="packages/web_components/webcomponents.min.js"></script>
    <script async src="packages/web_components/dart_support.js"></script>

But it would be nice if it could simplify that down to just:

    <script async src="test.dart.js"></script>

This would be a transform config option like this:

    - dart_to_js_script_rewriter:
         shim_shadow_dom: true

It would not load the two web_components files if the browser natively supports shadowDom, thus quickening initial load time.

**Comment by sethladd** [comment]
This transformer is static, in the sense that it's done at build time. Since you care about CSP, I'm guessing that you don't want the test.dart.js file to do browser feature detection and dynamically insert the webcomponents.min.js if required. That's how I'd do it, if I didn't care about CSP.

Another option, to speed up load time, is to concatenate all the JS files into a single file.

Have you profiled your load time? Where is the hot spot?

**Comment by monkeysecurity** [comment]
Good call on the CSP violation.  Concatenating all the JS files would slow down browsers that do support shadowDom, as they don't need the webcomponents package.

Looks like the best solution is to remove shadowDom https://github.com/Netflix/security_monkey/commit/014e11d625e9d7883bea81c64866aa0fc21a5e94 which then allows me to remove webcomponents.js and dart_support.js.

Also, webcomponents.js is not CSP compatible: https://github.com/webcomponents/webcomponentsjs/issues/188

It also appears that AngularDart ShadowDom components with cssUrl are not compatible with CSP: https://github.com/angular/angular.dart/issues/1666

The hotspot for my app is downloading main.dart.js, which went from 1.1MB to 1.7MB when I passed the CSP flag to the dart2js transformer.

Closing this issue now.
